### ISSUE TITLE
Missing Welcome and Transport Unknown

### ISSUE THREAD

**Issue by daryl3** [andyet/signalmaster#53]
Hello All,

I have an older version of Signalmaster working. We decided to update the Signalmaster software and we are now getting some errors.

****************************************************************************************
Installation Information:
Fresh Ubuntu Install
cd /opt
git clone https://github.com/andyet/signalmaster
cd signalmaster
npm install async; npm install node-uuid; npm install redis; npm install underscore; npm install precommit-hook; npm install getconfig; npm install yetify; npm install socket.io

Edit /opt/signalmaster/config/development.json
Removed logLevel 3 due to warning in starting.
Left stun server pointing at google.
Added my turn server.
Started Signalmaster: node server.js

End of Installation Information
****************************************************************************************

When I load the server page, I should see  "Welcome to socket.io". Instead I see a blank page with no html on the page.

When I load the server page, passing socket.io information, I should get authentication info.

Example:
https://signaling.simplewebrtc.com/
Welcome to socket.io.

Example 2:
https://signaling.simplewebrtc.com/socket.io/1/?t=1444403810714
5-bsFKOMpkbXfDhSgHFP:60:60:websocket,htmlfile,xhr-polling,jsonp-polling

So I have a few questions:
1. Why am I no longer getting the welcome message in Example 1?
2. Why am I getting the following error in Example 2? {"code":0,"message":"Transport unknown"}
3. Are you using the newest version of Signalmaster on the https://signaling.simplewebrtc.com/ server? If not, which version are you using?
4. What Linux flavor do you prefer? I tried Ubuntu and Redhat. Both had the same problem.

Thank you,

Daryl

**Comment by KevCoStudios** [comment]
I am having exactly the same issues and have the same questions.  I've tried multiple fresh installs of Ubuntu and Lubuntu, following the same procedure.  Is it an Ubuntu only issue?  I'm surprised this isn't a more popular thread.

**Comment by daryl3** [comment]
@KevCoStudios - So far, my guess is that not a lot of people have upgraded to the current version. Most folks I would imagine just leave their server running on the version they installed, until they have a good reason to upgrade.  I have also tried this on Redhat and had the same exact problems, so it is not just Ubuntu.

I wonder if @fippo has any thoughts on this?

**Comment by fippo** [comment]
this is probably related to the socket.io 1.x upgrade (#47) - make sure you are using the correct version

**Comment by daryl3** [comment]
@fippo Thanks, I believe the article you referenced is just saying to update to socket.io v.1.3.7, correct?  The package.json file included with the Signalmaster software is set to install socket.io v.1.3.7 automatically.

So I believe I am already doing what you are asking.  Unless you meant something else?  Here is what I have installed:
**************************************************************************************************************
root@signalmaster2:/opt/signalmaster# npm list socket.io
signal-master@0.2.3 /opt/signalmaster
└── socket.io@1.3.7
**************************************************************************************************************

**Comment by fippo** [comment]
err... correct version of simplewebrtc -- this plays well with 2.x

**Comment by daryl3** [comment]
@fippo Thanks, I am using the Simplewebrtc 2.x code. It looks like your answer is about something different then what I was asking. I'll try to rephrase my example/question. I am not asking about what the Simplewebrtc code sees, I am asking why the Signalmaster examples from a normal web browser call aren't working.

The README.md at https://github.com/andyet/signalmaster says that you should see a "Welcome to socket.io" message, if you go directly to your Signalmaster server in a web browser.

Example:
Go to https://signaling.simplewebrtc.com/ in a web browser, and you should see:
Welcome to socket.io.

Right now, a new Signalmaster install does not do this.  I'm not sure what changed, but this stopped working as the README.md suggests and as it worked in previous versions.

I also have an example of what you should see if you do an actual handshake.  If you go to https://signaling.simplewebrtc.com/socket.io/1/?t=1444403810714 in a web browser, you will see something similar to the following:
5-bsFKOMpkbXfDhSgHFP:60:60:websocket,htmlfile,xhr-polling,jsonp-polling

Instead, we see the following:
{"code":0,"message":"Transport unknown"}

I hope this clarification helps, please let me know if you need me to clarify or give any better examples.

Thanks,

Daryl

**Comment by fippo** [comment]
yeah, that is one one of the things that changed in socket.io. The README needs to be updated.

**Comment by daryl3** [comment]
This might also help, on the signalmaster server, it crashes and displays the following when you try to go to the "Welcome Page".

&yet -- signal master is running at: http://localhost:8888

crypto.js:226
  this._binding.init(hmac, toBuf(key));
                ^
TypeError: Not a buffer
    at new Hmac (crypto.js:226:17)
    at Object.Hmac (crypto.js:224:12)
    at /opt/signalmaster/sockets.js:118:35
    at Array.forEach (native)
    at Namespace.<anonymous> (/opt/signalmaster/sockets.js:117:32)
    at Namespace.EventEmitter.emit (events.js:95:17)
    at Namespace.emit (/opt/signalmaster/node_modules/socket.io/lib/namespace.js:205:10)
    at /opt/signalmaster/node_modules/socket.io/lib/namespace.js:172:14
    at process._tickCallback (node.js:415:13)

**Comment by fippo** [comment]
@daryl3 did you remove turn from the config or are missing a credential?

**Comment by daryl3** [comment]
Ok, I went ahead and tested it again. This time, I used the Newest SimpleWebRTC Code and the Newest Signal Master. The welcome screen does not work.  The old method of just passing a handshake test doesn't work.  However, the actual video tests do work.  So, it sounds like the README just needs to be updated, to no longer suggest that the welcome page should work.  And you might want to add a note that indicates it used to, but no longer does, for people with notes that are expecting that to be a valid test.

Thanks for the help @fippo
